drivers/i2s/i2s_stm32_sai.c

```diff
@@ -806,7 +816,7 @@ static DEVICE_API(i2s, i2s_stm32_driver_api) = {
 		.pclk_len = DT_INST_NUM_CLOCKS(index),                                             \
 		.pcfg = PINCTRL_DT_INST_DEV_CONFIG_GET(index),                                     \
 		.mclk_enable = DT_INST_PROP(index, mclk_enable),                                   \
-		.mclk_div_enable = DT_INST_PROP(index, mclk_div_enable),                           \
+		.mclk_div = (enum mclk_divider)DT_ENUM_IDX(DT_DRV_INST(index), mclk_divider),      \
```

Comment:
Nit: instead of the unstable `DT_ENUM_IDX`, something more stable like this could be used:
```suggestion
		.mclk_div = CONCAT(MCLK_, DT_INST_STRING_UPPER_TOKEN(index, mclk_divider)),      \
```